lib/Logger/LogMacros.h

```diff
@@ -58,16 +58,36 @@
 #include "Logger/Logger.h"
 #include "Logger/LoggerStream.h"
 
-////////////////////////////////////////////////////////////////////////////////
-/// @brief logs a message for a topic
-////////////////////////////////////////////////////////////////////////////////
 
 #define ARANGO_INTERNAL_LOG_HELPER(id)                        \
   ::arangodb::Logger::LINE(__LINE__)                          \
   << ::arangodb::Logger::FILE(__FILE__)                       \
   << ::arangodb::Logger::FUNCTION(__FUNCTION__)               \
   << ::arangodb::Logger::LOGID((id))
 
+#define ARANGODB_UNCONDITIONALLY_BUILD_LOG_MESSAGES 1
+#if ARANGODB_UNCONDITIONALLY_BUILD_LOG_MESSAGES
+// these versions *allways* generate the log string, only to discard it
+// just before logging them. This is good for ASAN or coverage.
+// DONT use in production.
+#define LOG_TOPIC(id, level, topic)                                     \
+  ::arangodb::LoggerStream() <<  (::arangodb::LogLevel::level)          \
+                             << (topic)                                 \
+                             << ARANGO_INTERNAL_LOG_HELPER(id)
+
+#define LOG_TOPIC_IF(id, level, topic, cond)                            \
+  false && (cond)                                                       \
```

Comment:
Now this will never execute `(cond)` which could again lower coverage etc.
It is also not safe to always generate the log message regardless of whether the condition is true.
Consider:
```
LOG_TOPIC_IF("abcde", WARN, Logger::FIXME, (result != nullptr)) << "RESULT: " << result->thisWillCrashNow();
```

```suggestion
  !(cond)                                                       \
```